src/framework/handlers/script.js

```diff
@@ -97,28 +100,65 @@ class ScriptHandler {
     patch(asset, assets) { }
 
     _loadScript(url, callback) {
-        const head = document.head;
-        const element = document.createElement('script');
-        this._cache[url] = element;
-
-        // use async=false to force scripts to execute in order
-        element.async = false;
-
-        element.addEventListener('error', function (e) {
-            callback(`Script: ${e.target.src} failed to load`);
-        }, false);
-
-        let done = false;
-        element.onload = element.onreadystatechange = function () {
-            if (!done && (!this.readyState || (this.readyState === 'loaded' || this.readyState === 'complete'))) {
-                done = true; // prevent double event firing
-                callback(null, url, element);
-            }
-        };
-        // set the src attribute after the onload callback is set, to avoid an instant loading failing to fire the callback
-        element.src = url;
 
-        head.appendChild(element);
+        const path = url.split('?')[0];
+        const isEsmScript = path.endsWith('.esm.js');
```

Comment:
wondering why not `.mjs`?

Reply:
Open to suggestion, however one thing the V8 team and Mozilla pointed out is that '.js' will get served with the correct Content-Type headers that would already exist on the server. 'mjs', 'jsm' etc may not have those configured and would need to be manually setup . https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules#aside_%E2%80%94_.mjs_versus_.js

Reply:
Although build systems, today do support mjs pretty well, e.g. Babel.
We do have page that does require setting up server already: https://developer.playcanvas.com/en/user-manual/publishing/web/self-hosting-for-beginners/#content-types
It worth adding there configs too.

Reply:
That's true. I'm not particularly wedded to either. Currently the editor doesn't let you change the file suffix, so that would need to be changed, and I'm not sure what it reall buys you. The nice thing about keeping it '.js' is that the user can just change directly as-is. Also one other thing, there's gonna be a distinction between ScriptType modules, and just plain vanilla modules (config/utils/etc). Arguably these shouldn't __only__ work if they're '.mjs'. Also potentially, if we enforced the ESM ScriptType at the file name level then we can guard against it being added as a script in the editor, rather than having to load it and check through exports.